.github/workflows/linux.yml

```diff
@@ -16,476 +16,490 @@ concurrency:
-        run: |
-          ${OV_INSTALL_DIR}/samples/cpp/build_samples.sh -i ${{ github.workspace }}/s\ pace
-
-      - name: Build samples (Debug)
-        if: ${{ 'Release' != matrix.build-type }}
-        run: |
-          source ${OV_INSTALL_DIR}/setupvars.sh
-          cmake -DCMAKE_BUILD_TYPE=${{ matrix.build-type }} -S ${OV_INSTALL_DIR}/samples/cpp/ -B ./samples\ build/ && cmake --build ./samples\ build/ --config ${{ matrix.build-type }} -j
-          cmake --install ./samples\ build/ --config ${{ matrix.build-type }} --component samples_bin --prefix s\ pace
-
-      - name: Test C++ samples (greedy_causal_lm)
+          path: ${{ env.SRC_DIR }}
+          submodules: recursive
+          
+      - name: Download Build Artifacts
+        uses: actions/download-artifact@fa0a91b85d4f404e444e00e005971372dc801d16 # v4.1.8
+        with:
+          pattern: "{${{ needs.openvino_download.outputs.ov_artifact_name }},ov_samples_${{ matrix.build-type }},genai_archive_${{ matrix.build-type }},genai_samples_${{ matrix.build-type }},genai_wheels}"
+          path: ${{ env.INSTALL_DIR }}
+          merge-multiple: true
+      
+      - name: Extract Artifacts
         run: |
-          source ${OV_INSTALL_DIR}/setupvars.sh
-          timeout 25s ${{ github.workspace }}/s\ pace/samples_bin/greedy_causal_lm ./TinyLlama-1.1B-Chat-v1.0/ ""
-
-      - name: Test C++ samples (whisper_speech_recognition)
+          pigz -dc ${{ env.GENAI_ARCHIVE_NAME }} | tar -xf - -C ${{ env.INSTALL_DIR }}
+          pigz -dc ${{ env.GENAI_SAMPLES_NAME }} | tar -xf - -C ${{ env.INSTALL_DIR }}
+          mkdir -p ${{ env.INSTALL_SECONDARY_DIR }}
+          pigz -dc ${{ env.OV_SAMPLES_NAME }} | tar -xf - -C ${{ env.INSTALL_SECONDARY_DIR }}
+        working-directory: ${{ env.INSTALL_DIR }}
+        
+      - name: Install GenAI Wheels
+        run: python -m pip install ${{ env.INSTALL_DIR }}/wheels/openvino_genai*.whl openvino-tokenizers[transformers] ${{ needs.openvino_download.outputs.ov_wheel_source }} --force-reinstall
+        working-directory: ${{ env.INSTALL_DIR }}
+     
+      - name: Install Samples Requirements
+        run: python -m pip install -r ./samples/requirements.txt
+        working-directory: ${{ env.SRC_DIR }}
+     
+      - name: Tests Prerequisites
         run: |
-          source ${OV_INSTALL_DIR}/setupvars.sh
-          wget https://storage.openvinotoolkit.org/models_contrib/speech/2021.2/librispeech_s5/how_are_you_doing_today.wav
-          timeout 25s ${{ github.workspace }}/s\ pace/samples_bin/whisper_speech_recognition ./whisper-tiny/ how_are_you_doing_today.wav
+          mkdir -p ${{ env.MODELS_DIR }}
+          optimum-cli export openvino --trust-remote-code --model TinyLlama/TinyLlama-1.1B-Chat-v1.0 ${{ env.MODELS_DIR }}/TinyLlama-1.1B-Chat-v1.0
+          optimum-cli export openvino --trust-remote-code --model openai/whisper-tiny ${{ env.MODELS_DIR }}/whisper-tiny
```

Comment:
is it possible to call these `optimum-cli` in parallel? they can interleave stages of model download / model conversion. 

Also, it's a good approach to use hf-transfer https://pypi.org/project/hf-transfer/ to speed-up models download.
Note, it requires `export HF_HUB_ENABLE_HF_TRANSFER=1`

Reply:
It is not a part of this PR, let's make it separate - https://github.com/openvinotoolkit/openvino.genai/pull/1300